pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -188,9 +190,15 @@ func getVirtualMachineByName(cfg *VSphereConfig, ctx context.Context, c *govmomi
 	}
 	f.SetDatacenter(dc)
 
+	var workingDir string
+	if len(cfg.Global.WorkingDir) > 0 {
+		workingDir = path.Clean(cfg.Global.WorkingDir) + "/"
```

Comment:
you can move the check to newVSphere() which gets called during init to avoid duplication.